connector/spanmetricsconnectorv2/connector.go

```diff
@@ -88,3 +92,58 @@ func (sm *spanMetrics) ConsumeTraces(ctx context.Context, td ptrace.Traces) erro
 	}
 	return sm.next.ConsumeMetrics(ctx, processedMetrics)
 }
+
+// calculateAdjustedCount calculates the adjusted count which represents
+// the number of spans in the population that are represented by the
+// individually sampled span. If the span is not-sampled OR if a non-
+// probability sampler is used then adjusted count defaults to 1.
+// https://opentelemetry.io/docs/specs/otel/trace/tracestate-probability-sampling/#adjusted-count
+func calculateAdjustedCount(tracestate string) uint64 {
+	w3cTraceState, err := sampling.NewW3CTraceState(tracestate)
+	if err != nil {
+		return 1
+	}
+	otTraceState := w3cTraceState.OTelValue()
+	if otTraceState == nil {
+		return 1
+	}
+	// For probability sampler, calculate the adjusted count based on t-value (`th`).
+	if len(otTraceState.TValue()) != 0 {
+		// TODO (lahsivjar): Should we handle fractional adjusted count?
+		// One way to do this would be to scale the values in the histograms
+		// for some precision.
+		return uint64(otTraceState.AdjustedCount())
+	}
+	// For consistent probablity sampler, calculate the adjusted count based on
+	// p-value, negative base-2 logarithm of sampling probability
+	var p uint64
+	for _, kv := range otTraceState.ExtraValues() {
+		if kv.Key == "p" {
+			// If p-value is present then calculate the adjusted count as per
+			// the consistent probability sampling specs.
+			if kv.Value != "" {
+				// p-value is represented as unsigned decimal integers
+				// requiring at most 6 bits of information. We parse to
+				// 7 bits as 63 holds a special meaning and thus needs
+				// to be distinguished w.r.t. other invalid >63 values.
+				p, _ = strconv.ParseUint(kv.Value, 10, 7)
+			}
+			break
+		}
+	}
+	switch {
+	case p == 0:
+		return 1
+	case p == 63:
+		// p-value == 63 represents zero adjusted count
+		return 0
+	case p > 63:
+		// Invalid value, default to 1
+		return 1
+	default:
+		// TODO (lahsivjar): Should we handle fractional adjusted count?
+		// One way to do this would be to scale the values in the histograms
+		// for some precision.
+		return uint64(math.Pow(2, float64(p)))
+	}
```

Comment:
I'm not sure about this...

My understanding is that the old (experimental) p-value/r-value spec is superseded by OTEP 235, which introduces t-value.

If we need to support both, perhaps it should be a separate processor that transforms the old tracestate? We might want to do something similar to handle the Elastic APM tracestate.

Reply:
> My understanding is that the old (experimental) p-value/r-value spec is superseded by OTEP 235, which introduces t-value.

+1. I didn't know that [probabilistic sampling processor already supported the new specs](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31894) and wanted to continue supporting that processor.

> If we need to support both, perhaps it should be a separate processor that transforms the old tracestate? We might want to do something similar to handle the Elastic APM tracestate.

This makes more sense. I have updated the PR to remove the p-value handling in favor of OTEP 235.